backend/src/quoteServer/quoteServer.py

```diff
@@ -0,0 +1,77 @@
+""" Used to query quotes from the quote server
+"""
+""" TODO: 
+        - test with workload generator
+""" 
+
+import socket
+import sys
+
+QUOTE_PORT       = 4444
+QUOTE_HOST_NAME  = '192.168.4.2'
+
+PACKET_SIZE 	 = 1024
+
+class QuoteServer:
+	def __init__(self, hostname=QUOTE_HOST_NAME, port=QUOTE_PORT,):
+		
+		self.hostname = hostname
+		self.port     = port
+
+	def connect(self):
+		"""
+		Connect to quote server.
+		"""
+		self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+
+		try:
+			print(f"Connecting to {self.hostname}:{self.port} ...")
+			self.socket.connect((self.hostname, self.port))
+		except socket.error as err:
+			print(f"ERROR! binding port {self.port} failed with error: {err}")
+			self.socket.close()
+			sys.exit(1)
+
+	def getQuote(self, cmdDict):
+		"""
+			Get a quote from the quote server
+		"""
+
+		# Get query and send to quote server
+		quote =  cmdDict['stockSymbol'] + ',' + cmdDict['user'] + "\n"
+			
+		self.socket.send(quote.encode())
```

Comment:
I think from the comments in the tutorial we need to reinitialize the socket each time we have a query request.

Reply:
Yup, I'll do that.